File: api/python/Grpcs/TextFilterGrpcAsyncProvider.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-
# ImageFilterAsyncProvider.py
# 2021, Lin Zhijun, https://github.com/toolgood/ToolGood.TextFilter.Api
# MIT Licensed 

import grpc
from Grpcs.GrpcBase import textFilter_pb2
from Grpcs.GrpcBase import textFilter_pb2_grpc


class TextFilterGrpcAsyncProvider():
    def __init__(self,textFilterConfig):
        self.__textFilterConfig=textFilterConfig

    def TextFilter(self,txt:str,skipBidi:bool,onlyPosition:bool,requestId:str,url:str):
        """
        @txt 需要检测的文本

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字
        """
        channel = grpc.insecure_channel(self.__textFilterConfig.GetGrpcHost())
        stub = textFilter_pb2_grpc.TextFilterGrpcStub(channel)
        response = stub.TextFilterAsync(textFilter_pb2.TextFindAllAsyncGrpcRequest(txt=txt, skipBidi=skipBidi,onlyPosition=onlyPosition,requestId=requestId,url=url))
        return response


    def TextReplace(self,txt:str,replaceChar:str,reviewReplace:bool,contactReplace:bool,skipBidi:bool,onlyPosition:bool,requestId:str,url:str):
        """
        @txt 需要检测的文本

        @replaceChar 替换符号, 默认 星号

        @reviewReplace 人工审核替换，默认true

        @contactReplace 联系字符串替换，默认true

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        """
        channel = grpc.insecure_channel(self.__textFilterConfig.GetGrpcHost())
        stub = textFilter_pb2_grpc.TextFilterGrpcStub(channel)
        replaceCh=ord('*')
        if len(replaceChar)>0:
            replaceCh=ord(replaceChar[0])
        response = stub.TextReplaceAsync(textFilter_pb2.TextReplaceAsyncGrpcRequest(txt=txt,replaceChar=replaceCh,reviewReplace=reviewReplace,contactReplace=contactReplace, skipBidi=skipBidi,onlyPosition=onlyPosition,requestId=requestId,url=url))
        return response

    def HtmlFilter(self,txt:str,skipBidi:bool,onlyPosition:bool,requestId:str,url:str):
        """
        @txt 需要检测的文本

        @skipBidi 是否跳过 Bidi 字符，默认 false
        
        @onlyPosition 只显示位置，不显示匹配文字
        """
        channel = grpc.insecure_channel(self.__textFilterConfig.GetGrpcHost())
        stub = textFilter_pb2_grpc.TextFilterGrpcStub(channel)
        response = stub.HtmlFilterAsync(textFilter_pb2.TextFindAllAsyncGrpcRequest(txt=txt, skipBidi=skipBidi,onlyPosition=onlyPosition,requestId=requestId,url=url))
        return response
 

    def HtmlReplace(self,txt:str,replaceChar:str,reviewReplace:bool,contactReplace:bool,skipBidi:bool,onlyPosition:bool,requestId:str,url:str):
        """
        @txt 需要检测的文本

        @replaceChar 替换符号, 默认 星号

        @reviewReplace 人工审核替换，默认true

        @contactReplace 联系字符串替换，默认true

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        """
        channel = grpc.insecure_channel(self.__textFilterConfig.GetGrpcHost())
        stub = textFilter_pb2_grpc.TextFilterGrpcStub(channel)
        replaceCh=ord('*')
        if len(replaceChar)>0:
            replaceCh=ord(replaceChar[0])
        response = stub.HtmlReplaceAsync(textFilter_pb2.TextReplaceAsyncGrpcRequest(txt=txt,replaceChar=replaceCh,reviewReplace=reviewReplace,contactReplace=contactReplace, skipBidi=skipBidi,onlyPosition=onlyPosition,requestId=requestId,url=url))
        return response

    def JsonFilter(self,txt:str,skipBidi:bool,onlyPosition:bool,requestId:str,url:str):
        """
        @txt 需要检测的文本

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字
        """
        channel = grpc.insecure_channel(self.__textFilterConfig.GetGrpcHost())
        stub = textFilter_pb2_grpc.TextFilterGrpcStub(channel)
        response = stub.JsonFilterAsync(textFilter_pb2.TextFindAllAsyncGrpcRequest(txt=txt, skipBidi=skipBidi,onlyPosition=onlyPosition,requestId=requestId,url=url))
        return response

    def JsonReplace(self,txt:str,replaceChar:str,reviewReplace:bool,contactReplace:bool,skipBidi:bool,onlyPosition:bool,requestId:str,url:str):
        """
        @txt 需要检测的文本

        @replaceChar 替换符号, 默认 星号

        @reviewReplace 人工审核替换，默认true

        @contactReplace 联系字符串替换，默认true

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        """
        channel = grpc.insecure_channel(self.__textFilterConfig.GetGrpcHost())
        stub = textFilter_pb2_grpc.TextFilterGrpcStub(channel)
        replaceCh=ord('*')
        if len(replaceChar)>0:
            replaceCh=ord(replaceChar[0])
        response = stub.JsonReplaceAsync(textFilter_pb2.TextReplaceAsyncGrpcRequest(txt=txt,replaceChar=replaceCh,reviewReplace=reviewReplace,contactReplace=contactReplace, skipBidi=skipBidi,onlyPosition=onlyPosition,requestId=requestId,url=url))
        return response
 
    def MarkdownFilter(self,txt:str,skipBidi:bool,onlyPosition:bool,requestId:str,url:str):
        """
        @txt 需要检测的文本

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字
        """
        channel = grpc.insecure_channel(self.__textFilterConfig.GetGrpcHost())
        stub = textFilter_pb2_grpc.TextFilterGrpcStub(channel)
        response = stub.MarkdownFilterAsync(textFilter_pb2.TextFindAllAsyncGrpcRequest(txt=txt, skipBidi=skipBidi,onlyPosition=onlyPosition,requestId=requestId,url=url))
        return response

    def MarkdownReplace(self,txt:str,replaceChar:str,reviewReplace:bool,contactReplace:bool,skipBidi:bool,onlyPosition:bool,requestId:str,url:str):
        """
        @txt 需要检测的文本

        @replaceChar 替换符号, 默认 星号

        @reviewReplace 人工审核替换，默认true

        @contactReplace 联系字符串替换，默认true

        @skipBidi 是否跳过 Bidi 字符，默认 false

        @onlyPosition 只显示位置，不显示匹配文字

        """
        channel = grpc.insecure_channel(self.__textFilterConfig.GetGrpcHost())
        stub = textFilter_pb2_grpc.TextFilterGrpcStub(channel)
        replaceCh=ord('*')
        if len(replaceChar)>0:
            replaceCh=ord(replaceChar[0])
        response = stub.MarkdownReplaceAsync(textFilter_pb2.TextReplaceAsyncGrpcRequest(txt=txt,replaceChar=replaceCh,reviewReplace=reviewReplace,contactReplace=contactReplace, skipBidi=skipBidi,onlyPosition=onlyPosition,requestId=requestId,url=url))
        return response

File: api/golang/ToolGood/TextFilter/Api/Grpcs/TextFilterGrpcAsyncProvider.go
package Grpcs

import (
	"log"

	pb "./GrpcBase"
	"golang.org/x/net/context"
	"google.golang.org/grpc"

	. "../Datas"
	. "../Datas/Requests"
)

type TextFilterGrpcAsyncProvider struct {
	grpcHost string
}

func NewTextFilterGrpcAsyncProvider(grpcHost string) *TextFilterGrpcAsyncProvider {
	result := &TextFilterGrpcAsyncProvider{}
	result.grpcHost = grpcHost
	return result
}

func (this *TextFilterGrpcAsyncProvider) TextFilter(request *TextFilterAsyncRequest) *CommonResult {
	conn, err := grpc.Dial(this.grpcHost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	t := pb.NewTextFilterGrpcClient(conn)

	reqBody := new(pb.TextFindAllAsyncGrpcRequest)
	reqBody.Txt = request.Txt
	reqBody.SkipBidi = request.SkipBidi
	reqBody.OnlyPosition = request.OnlyPosition

	tr, err := t.TextFilterAsync(context.Background(), reqBody)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
		return nil
	}
	result := this.createCommonResult(tr)
	defer conn.Close()
	return result
}
func (this *TextFilterGrpcAsyncProvider) HtmlFilter(request *TextFilterAsyncRequest) *CommonResult {
	conn, err := grpc.Dial(this.grpcHost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	t := pb.NewTextFilterGrpcClient(conn)

	reqBody := new(pb.TextFindAllAsyncGrpcRequest)
	reqBody.Txt = request.Txt
	reqBody.SkipBidi = request.SkipBidi
	reqBody.OnlyPosition = request.OnlyPosition

	tr, err := t.HtmlFilterAsync(context.Background(), reqBody)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
		return nil
	}
	result := this.createCommonResult(tr)
	defer conn.Close()
	return result
}
func (this *TextFilterGrpcAsyncProvider) JsonFilter(request *TextFilterAsyncRequest) *CommonResult {
	conn, err := grpc.Dial(this.grpcHost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	t := pb.NewTextFilterGrpcClient(conn)

	reqBody := new(pb.TextFindAllAsyncGrpcRequest)
	reqBody.Txt = request.Txt
	reqBody.SkipBidi = request.SkipBidi
	reqBody.OnlyPosition = request.OnlyPosition

	tr, err := t.JsonFilterAsync(context.Background(), reqBody)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
		return nil
	}
	result := this.createCommonResult(tr)
	defer conn.Close()
	return result
}
func (this *TextFilterGrpcAsyncProvider) MarkdownFilter(request *TextFilterAsyncRequest) *CommonResult {
	conn, err := grpc.Dial(this.grpcHost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	t := pb.NewTextFilterGrpcClient(conn)

	reqBody := new(pb.TextFindAllAsyncGrpcRequest)
	reqBody.Txt = request.Txt
	reqBody.SkipBidi = request.SkipBidi
	reqBody.OnlyPosition = request.OnlyPosition

	tr, err := t.MarkdownFilterAsync(context.Background(), reqBody)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
		return nil
	}
	result := this.createCommonResult(tr)
	defer conn.Close()
	return result
}

func (this *TextFilterGrpcAsyncProvider) TextReplace(request *TextReplaceRequest) *CommonResult {
	conn, err := grpc.Dial(this.grpcHost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	t := pb.NewTextFilterGrpcClient(conn)

	reqBody := new(pb.TextReplaceAsyncGrpcRequest)
	reqBody.Txt = request.Txt
	reqBody.ReplaceChar = 42 //*
	if len(request.ReplaceChar) > 0 {
		byteArray := []rune(request.ReplaceChar)
		reqBody.ReplaceChar = uint32(byteArray[0])
	}
	reqBody.ReviewReplace = request.ReviewReplace
	reqBody.ContactReplace = request.ContactReplace
	reqBody.SkipBidi = request.SkipBidi
	reqBody.OnlyPosition = request.OnlyPosition

	tr, err := t.TextReplaceAsync(context.Background(), reqBody)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
		return nil
	}
	result := this.createCommonResult(tr)
	defer conn.Close()
	return result
}

func (this *TextFilterGrpcAsyncProvider) HtmlReplace(request *TextReplaceRequest) *CommonResult {
	conn, err := grpc.Dial(this.grpcHost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	t := pb.NewTextFilterGrpcClient(conn)

	reqBody := new(pb.TextReplaceAsyncGrpcRequest)
	reqBody.Txt = request.Txt
	reqBody.ReplaceChar = 42 //*
	if len(request.ReplaceChar) > 0 {
		byteArray := []rune(request.ReplaceChar)
		reqBody.ReplaceChar = uint32(byteArray[0])
	}
	reqBody.ReviewReplace = request.ReviewReplace
	reqBody.ContactReplace = request.ContactReplace
	reqBody.SkipBidi = request.SkipBidi
	reqBody.OnlyPosition = request.OnlyPosition

	tr, err := t.HtmlReplaceAsync(context.Background(), reqBody)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
		return nil
	}
	result := this.createCommonResult(tr)
	defer conn.Close()
	return result
}

func (this *TextFilterGrpcAsyncProvider) JsonReplace(request *TextReplaceRequest) *CommonResult {
	conn, err := grpc.Dial(this.grpcHost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	t := pb.NewTextFilterGrpcClient(conn)

	reqBody := new(pb.TextReplaceAsyncGrpcRequest)
	reqBody.Txt = request.Txt
	reqBody.ReplaceChar = 42 //*
	if len(request.ReplaceChar) > 0 {
		byteArray := []rune(request.ReplaceChar)
		reqBody.ReplaceChar = uint32(byteArray[0])
	}
	reqBody.ReviewReplace = request.ReviewReplace
	reqBody.ContactReplace = request.ContactReplace
	reqBody.SkipBidi = request.SkipBidi
	reqBody.OnlyPosition = request.OnlyPosition

	tr, err := t.JsonReplaceAsync(context.Background(), reqBody)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
		return nil
	}
	result := this.createCommonResult(tr)
	defer conn.Close()
	return result
}

func (this *TextFilterGrpcAsyncProvider) MarkdownReplace(request *TextReplaceRequest) *CommonResult {
	conn, err := grpc.Dial(this.grpcHost, grpc.WithInsecure())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	t := pb.NewTextFilterGrpcClient(conn)

	reqBody := new(pb.TextReplaceAsyncGrpcRequest)
	reqBody.Txt = request.Txt
	reqBody.ReplaceChar = 42 //*
	if len(request.ReplaceChar) > 0 {
		byteArray := []rune(request.ReplaceChar)
		reqBody.ReplaceChar = uint32(byteArray[0])
	}
	reqBody.ReviewReplace = request.ReviewReplace
	reqBody.ContactReplace = request.ContactReplace
	reqBody.SkipBidi = request.SkipBidi
	reqBody.OnlyPosition = request.OnlyPosition

	tr, err := t.MarkdownReplaceAsync(context.Background(), reqBody)
	if err != nil {
		log.Fatalf("did not connect: %v", err)
		return nil
	}
	result := this.createCommonResult(tr)
	defer conn.Close()
	return result
}

func (this *TextFilterGrpcAsyncProvider) createCommonResult(response *pb.TextFilterRequestIdGrpcReply) *CommonResult {
	result := &CommonResult{}
	result.Code = response.GetCode()
	result.RequestId = response.GetRequestId()
	result.Message = response.GetMessage()

	return result
}
